extensions/cms/src/cmsUtils.ts

```diff
@@ -79,40 +105,65 @@ export class CmsUtils {
+		return (connectionA !== connectionB) || ((connectionA.connectionId === connectionB.connectionId) &&
+			(connectionA.options.savePassword !== connectionA.options.savePassword));
+	}
+
 	public async createCmsServer(connection: azdata.connection.Connection,
-		name: string, description: string): Promise<mssql.ListRegisteredServersResult> {
+		name: string, description: string): Promise<CreateCmsResult> {
 		let provider = await this.getCmsService();
 		connection.providerName = connection.providerName === cmsProvider ? mssqlProvider : connection.providerName;
 		let ownerUri = await azdata.connection.getUriForConnection(connection.connectionId);
 		if (!ownerUri) {
 			// Make a connection if it's not already connected
-			await azdata.connection.connect(Utils.toConnectionProfile(connection), false, false).then(async (result) => {
+			let initialConnectionProfile = CmsUtils.getConnectionProfile(connection);
+			await azdata.connection.connect(initialConnectionProfile, false, false).then(async (result) => {
 				ownerUri = await azdata.connection.getUriForConnection(result.connectionId);
+				// If the ownerUri is still undefined, then open a connection dialog with the connection
+				if (!ownerUri) {
+					await this.connection(initialConnectionProfile).then(async (result) => {
+						if (result) {
+							ownerUri = await azdata.connection.getUriForConnection(result.connectionId);
+							connection = result;
+						}
+					}, (error) => {
+						// cancel pressed on connection dialog
+						return Promise.reject(localize('cms.error.cancelConnectionDialog', 'The server is disconnected. Please connect to the server to continue.'));
+					});
+				}
 			});
 		}
 		return provider.createCmsServer(name, description, connection, ownerUri).then((result) => {
 			if (result) {
-				return Promise.resolve(result);
-			} else {
-				return Promise.reject(null);
+				let createCmsResult: CreateCmsResult = {
+					listRegisteredServersResult: result,
+					connection: connection
+				};
+				return Promise.resolve(createCmsResult);
 			}
+		}, (error) => {
+			return Promise.reject(error.message);
 		});
 	}
 
-	public async deleteCmsServer(cmsServer: any): Promise<void> {
+	public async deleteCmsServer(cmsServerName: string, connection: azdata.connection.Connection): Promise<void> {
 		let config = this.getConfiguration();
 		if (config && config.servers) {
 			let newServers = config.servers.filter((cachedServer) => {
-				return cachedServer.name !== cmsServer;
+				return cachedServer.name !== cmsServerName;
 			});
 			await this.setConfiguration(newServers);
 			this._registeredCmsServers = this._registeredCmsServers.filter((cachedServer) => {
-				return cachedServer.name !== cmsServer;
+				return cachedServer.name !== cmsServerName;
 			});
 		}
+		if (connection.options.authenticationType === 'SqlLogin' && connection.options.savePassword) {
```

Comment:
>'SqlLogin' [](start = 48, length = 10)

These values shouldn't be hardcoded like this - put them in a central place like constants.ts we have in the main ADS code